client/packet_sock_linux.go

```diff
@@ -0,0 +1,66 @@
+package client
+
+import (
+	"net"
+	"syscall"
+	"time"
+)
+
+// TODO: Make packetSock implement net.PacketConn?
+type packetSock struct {
+	fd      int
+	ifindex int
+}
+
+// NewPacketSock creates a new socket that sends and receives packets.
+func NewPacketSock(ifindex int) (*packetSock, error) {
```

Comment:
If it ends up being necessary to work at the link layer: https://github.com/mdlayher/raw.

Also supports BSD.